src/Var.cc

```diff
@@ -351,16 +445,24 @@ void begin_func(ID* id, const char* module_name, function_flavor flavor,
 		t->ClearYieldType(flavor);
 		}
 
+	std::optional<FuncType::Prototype> prototype;
+
 	if ( id->Type() )
 		{
-		if ( ! same_type(id->Type(), t.get()) )
-			id->Type()->Error("incompatible types", t.get());
+		prototype = func_type_check(id->Type()->AsFuncType(), t.get());
 
-		else
+		if ( prototype )
+			{
 			// If a previous declaration of the function had &default params,
```

Comment:
Does this derive default arguments only from the canonical version? (Which would seem like the right behaviour)

This is something worth both documenting and testing I think, mind adding that?

Reply:
Yeah, `&default` args only come through via the canonical event/hook prototype.  Added docs/tests and found a few other places that warranted additional checks.  Let me know if you look and think of anything else or want me to go ahead and merge.